tf/main.tf

```diff
@@ -0,0 +1,89 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+provider "ec" {}
+
+data "ec_stack" "latest" {
+  version_regex = "latest"
+  region        = var.ec_region
+}
+
+resource "ec_deployment" "ec_aws_lambda_minimal" {
+  name                   = "aws-lambda-smoke-testing-deployment"
+  region                 = var.ec_region
+  version                = data.ec_stack.latest.version
+  deployment_template_id = var.ec_deployment_template
+
+  elasticsearch {
+    autoscale = "false"
+
+    topology {
+      id         = "hot_content"
+      size       = "1g"
+      zone_count = 1
+    }
+  }
+}
```

Comment:
I think it's overkill to create a separate observability deployment for smoke tests. In the future we'll be able to point a deployment at itself for observability, if we need this: https://github.com/elastic/terraform-provider-ec/pull/502 (not yet released)

Reply:
Agree, should we wait for the next release of the terraform provider or pitch someone about it ?

Reply:
I would suggest we just remove it for now, and consider adding it back later when "self" is an option. I don't think it's needed for smoke testing, is it?

Reply:
Solved by switching to custom APM server module!